### ISSUE TITLE
Usage of python's logging module in opinionated

### ISSUE THREAD

**Issue by Nicholas-Mitchell** [isl-org/Open3D-ML#330]
## First issue

I think there is an issue with your logging setup, more specifically how you use [`LogRecord`](https://docs.python.org/3/library/logging.html#logging.setLogRecordFactory). You define your own in [`Open3D-ML/ml3d/utils/log.py`](https://github.com/isl-org/Open3D-ML/blob/200bee315a55772a9b40f1304d9165242f896786/ml3d/utils/log.py), which is used once in [`Open3D-ML/ml3d/tf/pipelines/object_detection.py`](https://github.com/isl-org/Open3D-ML/blob/200bee315a55772a9b40f1304d9165242f896786/ml3d/tf/pipelines/object_detection.py).

The real problem is that this is evaluated any time anything is imported from either the [torch](https://github.com/isl-org/Open3D-ML/blob/master/ml3d/torch/pipelines/__init__.py) **or** [tensorflow](https://github.com/isl-org/Open3D-ML/blob/master/ml3d/tf/pipelines/__init__.py) `pipelines`.

The specific effect is that is breaks the common logging usage because your `LogRecord` swallows the arguments to the log string formatter:

```python
import logging
log = getLogger(__name__)
log.warning("Something important: %s, "some helpful info")

# gives:
WARNING - 2021-08-01 00:48:22,341 - some_name - Something important: %s
# expected:
WARNING - 2021-08-01 00:48:22,341 - some_name - Something important: some helpful info
```

Because of your implementation, users would have to instead write:

```python
log.warning("Something important: {}", "some helpful info")   # This goes against basic python logging conventions.
```

## Second issue

When importing from many parts of Open3d-ML, the user's logging module setup is overridden, due to the way you use python's `logging` module.

This is because you do the following [along with your import statements](https://github.com/isl-org/Open3D-ML/search?q=logging.basicConfig):

```python
import logging

logging.basicConfig(level=logging.INFO)   # This line is the issue.
log = logging.getLogger(__name__)
```
Setting up the basic configuration should only be done inside an entry-point i.e. after `if __name__ == "__main__":`.

The way you do this, combined with the first issue above, means you often break the user's logging setup.

-------

I would classify these as bad practice for a library (not bugs exactly). Admittedly low priority, but annoying for the user ;)

I could maybe submit a PR to "fix" this, but don't know what knock-on effects it might have for other developers and users.

**Comment by sanskar107** [comment]
@Nicholas-Mitchell thanks for reporting. This is indeed a bad practice. This is a low priority for us right now, feel free to create a PR to fix this.

**Comment by Nicholas-Mitchell** [comment]
@sanskar107 - please take a look at the PR I submitted to fix this issue.

As mentioned in my original post above - I don't know what impact this might have on other users.

The `LogRecord` that you added ~1 year ago (git blame) is only used in the `{tf,torch}/pipelines/object_detection.py` and `{tf,torch}/pipelines/semantic_detection.py`. In these places, it seems to still not really be used as you the `LogRecord` was intended. e.g  [`log.info("Logging in file : {}".format(log_file_path))`](https://github.com/isl-org/Open3D-ML/blob/master/ml3d/tf/pipelines/semantic_segmentation.py#L153).

**Comment by LeKristapino** [comment]
This issue made us spend a day understanding why suddenly the logger went haywire :D. Is this released in v0.15.1?
